pinot-core/src/main/java/org/apache/pinot/core/segment/index/column/PhysicalColumnIndexContainer.java

```diff
@@ -124,14 +129,21 @@ public PhysicalColumnIndexContainer(SegmentDirectory.Reader segmentReader, Colum
         _invertedIndex =
             new BitmapInvertedIndexReader(segmentReader.getIndexFor(columnName, ColumnIndexType.INVERTED_INDEX),
                 metadata.getCardinality());
+        _rangeIndex = null;
```

Comment:
Not sure if the if-else blocks here can be simplified

Reply:
I hope so but dint want to do that as part of this PR.